version: '3.9'

x-common-env: &common-env
  environment:
    - POSTGRES_HOST=postgres
    - POSTGRES_PORT=5432
    - POSTGRES_DB=${POSTGRES_DB}
    - POSTGRES_USER=${POSTGRES_USER}
    - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    - PREFECT_API_DATABASE_CONNECTION_URL=postgresql+asyncpg://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/prefect

    - PGADMIN_DEFAULT_EMAIL=admin@example.com
    - PGADMIN_DEFAULT_PASSWORD=password
    - PGADMIN_CONFIG_SERVER_MODE=False

services:
  postgres:
    image: postgres:latest
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./docker/postgres/init.sql:/docker-entrypoint-initdb.d/init.sql
    <<: *common-env

  sca:
    image: matthieujln/velib:sca
    volumes:
      - ./data:/app/data
      - ./app/flows:/app/app/flows
    ports:
      - "4200:4200"
      - "8081:8080"  # Exposing the Prefect UI port
    command: sh -c "python main.py"
    depends_on:
      - postgres
    <<: *common-env

  pgadmin:
    image: dpage/pgadmin4
    volumes:
      - ./docker/pgadmin/servers.json:/pgadmin4/servers.json
    ports:
      - "5050:80"
    depends_on:
      - postgres
    <<: *common-env

volumes:
  postgres-data:
